nixos/modules/services/networking/headplane.nix

```
@@ -0,0 +1,439 @@
+    then
+      builtins.removeAttrs settingsWithoutNulls [ "oidc" ]
+    else
+      settingsWithoutNulls;
+  settingsFile = settingsFormat.generate "headplane-config.yaml" settingsWithoutEmptyOidc;
+in
+{
+  options.services.headplane = {
+    enable = mkEnableOption "Headplane";
+    package = mkPackageOption pkgs "headplane" { };
+
+    debug = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Enable debug logging.";
+    };
+
+    settings = mkOption {
+      description = ''
+        Headplane configuration options. Generates a YAML config file.
+        See: https://github.com/tale/headplane/blob/main/config.example.yaml
+      '';
+      type = types.submodule {
+        options = {
+          server = mkOption {
+            type = types.submodule {
+              options = {
+                host = mkOption {
+                  type = types.str;
+                  default = "127.0.0.1";
+                  description = "The host address to bind to.";
+                  example = "0.0.0.0";
+                };
+
+                port = mkOption {
+                  type = types.port;
+                  default = 3000;
+                  description = "The port to listen on.";
+                };
+
+                cookie_secret_path = mkOption {
+                  type = types.nullOr types.path;
+                  default = null;
+                  description = ''
+                    Path to a file containing the cookie secret.
+                    The secret must be exactly 32 characters long.
+                  '';
+                  example = "config.sops.secrets.headplane_cookie.path";
+                };
+
+                cookie_secure = mkOption {
+                  type = types.bool;
+                  default = true;
+                  description = ''
+                    Should the cookies only work over HTTPS?
+                    Set to false if running via HTTP without a proxy.
+                    Recommended to be true in production.
+                  '';
+                };
+
+                data_path = mkOption {
+                  type = types.path;
+                  default = "/var/lib/headplane";
+                  description = ''
+                    The path to persist Headplane specific data.
+                    All data going forward is stored in this directory, including the internal database and any cache related files.
+                    Data formats prior to 0.6.1 will automatically be migrated.
+                  '';
+                  example = "/var/lib/headplane";
+                };
+              };
+            };
+            default = { };
+            description = "Server configuration for Headplane web application.";
+          };
+
+          headscale = mkOption {
+            type = types.submodule {
+              options = {
+                url = mkOption {
+                  type = types.str;
+                  default = "http://127.0.0.1:8080";
```

If you run headplane, you probably also run headscale on the same machine, or at least make this conditional so that when headplane is enabled, set this as the default.

```suggestion
                  default = "http://127.0.0.1:${toString config.services.headscale.port}";
                  defaultText = lib.literalExpression ''http://127.0.0.1:''${toString config.services.headscale.port}'';
```